src/PrimaryGeneratorAction.cc

```diff
@@ -34,15 +34,27 @@ PrimaryGeneratorAction::PrimaryGeneratorAction(DetectorConstruction* pDetector)
 
 PrimaryGeneratorAction::~PrimaryGeneratorAction() { delete fParticleGun; }
 
-void PrimaryGeneratorAction::SetSpectrum(TH1D* spt, double eMin, double eMax) {
+void PrimaryGeneratorAction::SetSpectrum(TH1D* spt, double eMin, double eMax,
+                                         TString units = "NO_SUCH_PARA") {
     TString xLabel = (TString)spt->GetXaxis()->GetTitle();
 
-    if (xLabel.Contains("MeV")) {
-        energyFactor = 1.e3;
-    } else if (xLabel.Contains("GeV")) {
-        energyFactor = 1.e6;
+    // if units is not specified (""), then get units from the x axis of the histogram, otherwise use the
+    // units from the RML
```

Comment:
But the units of the RML are used to define the energy units of the range to be used. Not to define the units the user used at the TH1D histogram.

Perhaps one could add the option so that we are able to define the units of the TH1D x-axis.